nixos/modules/system/etc/setup-etc.py

```diff
@@ -0,0 +1,177 @@
+    either argument is a symlink to a file in /etc/static or a directory with
+    all children being static.
+    """
+
+    if path.is_symlink():
+        target = path.resolve()
+        return os.path.commonprefix([target, STATIC]) == STATIC
+
+    if path.is_dir():
+        return all(is_static(ent) for ent in path.iterdir())
+
+    return False
+
+def cleanup() -> None:
+    """
+    Remove dangling symlinks that point to /etc/static.  These are
+    configuration files that existed in a previous configuration but not
+    in the current one.  For efficiency, don't look under /etc/nixos
+    (where all the NixOS sources live).
+    """
+    for root, dirs, files in os.walk('/etc'):
+        if root == '/etc':
+            dirs.remove('nixos')
+
+        for f in files:
+            path = Path(root) / Path(f)
+            if path.is_symlink():
+                target = path.resolve()
+                if os.path.commonprefix([target, STATIC]) == STATIC:
+                    x = STATIC / path.relative_to('/etc')
+                    if x.is_symlink():
+                        print("removing obsolete symlink '{path}'".format(path=path),
+                              file=sys.stderr)
+                        path.unlink()
+
+def link_statics(etc: Path) -> Tuple[Set[Path], Set[Path]]:
+    """
+    For every file in the etc tree, create a corresponding symlink in
+    /etc to /etc/static.  The indirection through /etc/static is to make
+    switching to a new configuration somewhat more atomic.
+    """
+
+    clean = open(CLEAN_FILE, 'a')
+    created = set()
+    copied = set()
+    for root, dirs, files in os.walk(etc):
+        for f in files:
+            path = Path(root) / Path(f)
+            if os.path.commonprefix([etc, path]) != etc:
+                continue
+            fn = path.relative_to(etc)
+            target = Path("/etc") / f
+            target.parent.mkdir(parents=True)
+            created.add(fn)
+
+            # Rename doesn't work if target is directory.
+            if Path(f).is_symlink() and target.is_dir():
+                if is_static(target):
+                    try:
+                        shutil.rmtree(target)
+                    except IOError as exc:
+                        warn("error trying to remove '{target}': {exc}". \
+                             format(target=target, exc=exc))
+                else:
+                    warn("$target directory contains user files. Symlinking may fail.")
+
+            meta_f = Path("{f}.mode".format(f=f))
+            if meta_f.exists():
+                mode = open(meta_f, 'r').read().strip()
+                if mode == 'direct-symlink':
+                    src = (STATIC / fn).resolve()
+                    atomic_symlink_or_warn(src, target)
+                else:
+                    user = open("{f}.uid").read().strip()
+                    grp = open("{f}.gid").read().strip()
```

Comment:
It looks like `.format(f=f)` is missing.

Reply:
I think it should be rather `f"{f}.gid"`.